ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -282,6 +304,41 @@ boolean shutdownBlobStore(PartitionId id) {
     }
   }
 
+  /**
+   * Set the BlobStore stopped state with given {@link PartitionId} {@code id}.
+   * @param partitionIds a list of {@link PartitionId} of the {@link BlobStore} whose stopped state should be set.
+   * @param isStopped whether to mark BlobStore as stopped ({@code true}) or started.
+   * @return a list of {@link PartitionId} whose stopped state fails to be updated.
+   */
+  List<PartitionId> setBlobStoreStoppedState(List<PartitionId> partitionIds, boolean isStopped) {
+    Set<PartitionId> failToUpdateStores = new HashSet<>();
+    List<ReplicaId> replicasToUpdate = new ArrayList<>();
+    for (PartitionId id : partitionIds) {
+      BlobStore store = stores.get(id);
+      if (store == null) {
+        // no need to check if the store is started because this method could be called after store is successfully shutdown.
+        logger.error("store is not found on this disk when trying to update stoppedReplicas list");
+        failToUpdateStores.add(id);
+      } else {
+        replicasToUpdate.add(partitionToReplicaMap.get(id));
+      }
+    }
+    boolean updated = false;
+    if (replicaStatusDelegate != null) {
+      logger.trace("Setting replica stopped state via ReplicaStatusDelegate on replica {}",
+          Arrays.toString(replicasToUpdate.toArray()));
+      updated = isStopped ? replicaStatusDelegate.markStopped(replicasToUpdate)
+          : replicaStatusDelegate.unmarkStopped(replicasToUpdate);
+    } else {
+      logger.error("The ReplicaStatusDelegate is not instantiated");
```

Comment:
should this be an error (or warn)? Seems like warn to me

Reply:
done